src/mlpack/methods/ann/loss_functions/multilabel_softmargin_loss.hpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file multilabel_softmargin_loss.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Definition of the Multi Label Soft Margin Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+#define MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class MultiLabelSoftMarginLoss
+{
+ public:
+  /**
+   * Create the MultiLabelSoftMarginLoss object.
+	 *
+	 * @param weight A manual rescaling weight given to each class. Initialized to
+	 *        1 by default.
+	 * @param reduction Specifies the reduction to apply to the output. When true,
+	 * 				'mean' reduction is used, where sum of the output will be divided by
```

Comment:
Could you adjust these to keep them within 80 characters.
Also could you align them properly with the previous line.
Thanks.

Reply:
Yeah definitely. I don't know, something weird must have happened when saving the files locally most probably. Because, when I had written down these things, everything was according to the style format for other functions. 💁🏻

Anyway, thanks for pointing it out. I will correct the styling issues and will also include the tests in a commit within a couple of days.

Reply:
I figured out the reason for this happening. For the git diff that you are viewing in the browser, tab size is rendered as 8 spaces. Whereas locally for VS Code, I have the setting as tab size = 2 spaces. So, for me, it appeared as if all the indentations were perfect, but in the browser, everything got messed up.